pkg/envoy/cds/response.go

```diff
@@ -34,6 +29,33 @@ func NewResponse(meshCatalog catalog.MeshCataloger, proxy *envoy.Proxy, _ *xds_d
 		return nil, err
 	}
 
+	// Add an inbound prometheus cluster (from Prometheus to localhost)
+	if pod, err := envoy.GetPodFromCertificate(proxy.GetCertificateCommonName(), meshCatalog.GetKubeController()); err != nil {
+		log.Warn().Msgf("Could not find pod for connecting proxy %s. No metadata was recorded.", proxy.GetCertificateSerialNumber())
+	} else if meshCatalog.GetKubeController().IsMetricsEnabled(pod) {
+		clusters = append(clusters, getPrometheusCluster())
+	}
+
+	// Add an outbound tracing cluster (from localhost to tracing sink)
+	if cfg.IsTracingEnabled() {
+		clusters = append(clusters, getTracingCluster(cfg))
+	}
+
+	if proxy.Kind() == envoy.KindGateway {
+		// Build remote clusters based on allowed outbound services
+		for _, dstService := range meshCatalog.ListAllowedOutboundServicesForIdentity(proxyIdentity.ToServiceIdentity()) {
+			cluster, err := getPassthroughCluster(dstService)
+			if err != nil {
+				log.Error().Err(err).Msgf("Failed to construct service cluster for service %s for proxy with XDS Certificate SerialNumber=%s on Pod with UID=%s",
+					dstService.Name, proxy.GetCertificateSerialNumber(), proxy.String())
+				return nil, err
+			}
+
+			clusters = append(clusters, cluster)
+		}
+		return makeResources(proxy, clusters), nil
```

Comment:
Should leverage the `getUpstreamServiceCluster()` here if cluster discovery is happening through EDS.

Reply:
same commment as above